pkg/cmd/pr/checks/checks.go

```diff
@@ -83,163 +93,75 @@ func checksRun(opts *ChecksOptions) error {
 
 	isTerminal := opts.IO.IsStdoutTTY()
 
-	if opts.WebMode {
-		openURL := ghrepo.GenerateRepoURL(baseRepo, "pull/%d/checks", pr.Number)
-		if isTerminal {
-			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
-		}
-		return opts.Browser.Browse(openURL)
-	}
-
-	if len(pr.StatusCheckRollup.Nodes) == 0 {
-		return fmt.Errorf("no commit found on the pull request")
+	openURL := ghrepo.GenerateRepoURL(baseRepo, "pull/%d/checks", pr.Number)
+	if isTerminal {
+		fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
 	}
+	return opts.Browser.Browse(openURL)
+}
 
-	rollup := pr.StatusCheckRollup.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes
-	if len(rollup) == 0 {
-		return fmt.Errorf("no checks reported on the '%s' branch", pr.BaseRefName)
+func checksRunWatchMode(opts *ChecksOptions) error {
+	duration, err := time.ParseDuration(fmt.Sprintf("%ds", opts.Interval))
+	if err != nil {
+		return fmt.Errorf("could not parse interval: %w", err)
```

Comment:
I think we should move the parsing of the interval flag up into the `RunE` portion of the command. This is a direction we have been moving lately trying to create a consistency through commands of when/where to parse and deal with flag logic. Once this code moves we will need to change the type of `interval` from `int` to `Duration`. Lastly, since this is a flag parsing error we can use the `cmdutil.FlagErrorf` helper function for returning the error.

Reply:
Changed in 6e30b62a